docker/transformers-pytorch-deepspeed-amd-gpu/Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM rocm/pytorch:rocm5.7_ubuntu22.04_py3.10_pytorch_2.0.1
+LABEL maintainer="Hugging Face"
+
+ARG DEBIAN_FRONTEND=noninteractive
+ARG PYTORCH='2.0.1'
+ARG ROCM='5.7'
+
+RUN apt update && \
+    apt install -y --no-install-recommends libaio-dev git && \
+    apt clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN python3 -m pip install --no-cache-dir --upgrade pip
+
+RUN python3 -m pip uninstall -y apex
+
+ARG REF=main
+WORKDIR /
+RUN git clone https://github.com/huggingface/transformers && cd transformers && git checkout $REF
+RUN python3 -m pip install --no-cache-dir ./transformers[deepspeed-testing]
+
+# When installing in editable mode, `transformers` is not recognized as a package.
+# this line must be added in order for python to be aware of transformers.
+RUN cd transformers && python3 setup.py develop
+
+RUN python3 -c "from deepspeed.launcher.runner import main"
```

Comment:
This will (always) use the torch `2.0.1` which is not what we want. I suggest to 

- either based on `docker/transformers-pytorch-amd-gpu/Dockerfile` on current `main`
- or not even to build an AMD+deepespeed image: just build+install deepspeed at CI runtime
- otherwise, based on this image, but need to install the 

    - ARG PYTORCH='2.1.0'
    - ARG TORCH_VISION='0.16.0'
    - ARG TORCH_AUDIO='2.1.0'

Reply:
Thanks @ydshieh that makes sense, just upgraded the torch version in https://github.com/huggingface/transformers/pull/27633/commits/f846b80bed3b75471edfde5ae3b617d0b235765a. 
I cannot use `docker/transformers-pytorch-amd-gpu/Dockerfile` at the moment as there is an incompatibility with ROCm and deepspeed for some reason (coming from the adam extension), but can modify it and change its parent image if needed